intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/problems/ProblemLevelRemappingManager.kt

```diff
@@ -0,0 +1,99 @@
+package com.jetbrains.plugin.structure.intellij.problems
+
+import com.fasterxml.jackson.core.exc.StreamReadException
+import com.fasterxml.jackson.databind.DatabindException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.jetbrains.plugin.structure.base.problems.PluginProblem.Level.ERROR
+import com.jetbrains.plugin.structure.base.problems.PluginProblem.Level.WARNING
+import java.io.IOException
+import java.net.URL
+import kotlin.reflect.KClass
+
+interface ProblemLevelRemappingManager {
+  fun initialize(): LevelRemappingDefinitions
+}
+
+fun levelRemappingFromClassPathJson(): JsonUrlProblemLevelRemappingManager {
+  val pluginProblemsJsonUrl = JsonUrlProblemLevelRemappingManager::class.java.getResource(PLUGIN_PROBLEMS_FILE_NAME)
+    ?: throw IOException("Plugin problem level remapping definition cannot be found at <$PLUGIN_PROBLEMS_FILE_NAME>")
+  return JsonUrlProblemLevelRemappingManager(pluginProblemsJsonUrl)
+}
+
+private const val INTELLIJ_PROBLEMS_PACKAGE_NAME = "com.jetbrains.plugin.structure.intellij.problems"
+const val PLUGIN_PROBLEMS_FILE_NAME = "plugin-problems.json"
+
+class JsonUrlProblemLevelRemappingManager(private val pluginProblemsJsonUrl: URL) : ProblemLevelRemappingManager {
+  private val json = ObjectMapper()
```

Comment:
why not local in `load()`

Reply:
`ObjectMapper` seems quite a heavy object. There might be multiple invocations of `load`, so there is no need to create multiple instances.